libs/execution_context/ebpf_link.c

```diff
@@ -314,25 +336,95 @@ static ebpf_result_t
 _ebpf_link_instance_invoke(
     _In_ const void* extension_client_binding_context, _Inout_ void* program_context, _Out_ uint32_t* result)
 {
-    // No function entry exit traces as this is a high volume function.
+    ebpf_execution_context_state_t state = {0};
+    ebpf_result_t return_value;
+    return_value = _ebpf_link_instance_invoke_batch_begin(
+        extension_client_binding_context, sizeof(ebpf_execution_context_state_t), &state);
+
+    if (return_value != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    return_value = _ebpf_link_instance_invoke_batch(extension_client_binding_context, program_context, result, &state);
+    ebpf_assert(return_value == EBPF_SUCCESS);
+
+    return_value = _ebpf_link_instance_invoke_batch_end(extension_client_binding_context);
+    ebpf_assert(return_value == EBPF_SUCCESS);
+
+Done:
+    return return_value;
+}
+
+static ebpf_result_t
+_ebpf_link_instance_invoke_batch_begin(
+    _In_ const void* extension_client_binding_context, _In_ size_t state_size, _In_ void* state)
```

Comment:
Here SAL says `state` is `_In_`, but line 392 passes it to `ebpf_get_execution_context_state()` where it is marked as `_Out_`.  Hence I think this should be Out not In.
```suggestion
    _In_ const void* extension_client_binding_context, size_t state_size, _Out_ void* state)
```
Also don't use `_In_` with value types like `size_t`, just pointers.